Invalid template passed to ipfs object new [<template>] causes a sigsegv

<!-- Please report security issues by email to security@ipfs.io -->

#### Version information:

```
go-ipfs version: 0.7.0
Repo version: 10
System version: amd64/darwin
Golang version: go1.14.4
```

#### Description:

Running `ipfs object new` and not passing `unixfs-dir` as the template causes a big stack trace instead of just printing an error message:

```
$ ipfs object new hello
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x15107a7]

goroutine 1 [running]:
github.com/ipfs/go-blockservice.(*blockService).AddBlock(0xc000de3470, 0x0, 0x0, 0x0, 0x0)
	github.com/ipfs/go-blockservice@v0.1.3/blockservice.go:134 +0x37
github.com/ipfs/go-merkledag.(*dagService).Add(0xc0000e1410, 0x25d4840, 0xc0008328c0, 0x0, 0x0, 0x0, 0xc00095da90)
	github.com/ipfs/go-merkledag@v0.3.2/merkledag.go:51 +0x8c
github.com/ipfs/go-ipfs/core/coreapi.(*ObjectAPI).New(0xc0008a1b80, 0x25d4840, 0xc0008328c0, 0xc0001164a8, 0x1, 0x1, 0x0, 0xc000112fd0, 0x1a, 0x2191a26)
	github.com/ipfs/go-ipfs@v0.7.0/core/coreapi/object.go:53 +0xd3
github.com/ipfs/go-ipfs/core/commands/object.glob..func8(0xc0002ae000, 0x25d4cc0, 0xc0007808a0, 0x2023380, 0xc0007807e0, 0xc00091fc40, 0x1df067e)
	github.com/ipfs/go-ipfs@v0.7.0/core/commands/object/object.go:502 +0x1bf
github.com/ipfs/go-ipfs-cmds.(*executor).Execute(0xc0001809e8, 0xc0002ae000, 0x25d4cc0, 0xc0007808a0, 0x2023380, 0xc0007807e0, 0xc0007808a0, 0x0)
	github.com/ipfs/go-ipfs-cmds@v0.4.0/executor.go:77 +0x171
github.com/ipfs/go-ipfs-cmds/cli.Run(0x25d4840, 0xc0007fc940, 0x331be00, 0xc000186040, 0x4, 0x4, 0xc000180000, 0xc000180008, 0xc000180010, 0x23f4970, ...)
	github.com/ipfs/go-ipfs-cmds@v0.4.0/cli/run.go:137 +0xa72
main.mainRet(0x0)
	github.com/ipfs/go-ipfs@v0.7.0/cmd/ipfs/main.go:168 +0x451
main.main()
	github.com/ipfs/go-ipfs@v0.7.0/cmd/ipfs/main.go:71 +0x22
```

--- a/core/coreapi/object.go
+++ b/core/coreapi/object.go
@@ -48,6 +48,8 @@ func (api *ObjectAPI) New(ctx context.Context, opts ...caopts.ObjectNewOption) (
 		n = new(dag.ProtoNode)
 	case "unixfs-dir":
 		n = ft.EmptyDirNode()
+	default:
+		return nil, fmt.Errorf("unknown node type: %s", options.Type)
 	}
 
 	err = api.dag.Add(ctx, n)

--- a/test/sharness/t0051-object.sh
+++ b/test/sharness/t0051-object.sh
@@ -223,6 +223,10 @@ test_object_cmd() {
     ipfs object stat $OUTPUT
   '
 
+  test_expect_success "'ipfs object new foo' shouldn't crash" '
+    test_expect_code 1 ipfs object new foo
+  '
+
   test_expect_success "'ipfs object links' gives the correct results" '
     echo "$EMPTY_DIR" 4 foo > expected &&
     ipfs object links "$OUTPUT" > actual &&

